Mathlib/Algebra/Star/Unitary.lean

```diff
@@ -135,12 +136,36 @@ theorem _root_.IsUnit.mem_unitary_iff_mul_star_self {u : R} (hu : IsUnit u) :
 alias ⟨_, _root_.IsUnit.mem_unitary_of_star_mul_self⟩ := IsUnit.mem_unitary_iff_star_mul_self
 alias ⟨_, _root_.IsUnit.mem_unitary_of_mul_star_self⟩ := IsUnit.mem_unitary_iff_mul_star_self
 
+lemma mul_inv_mem_unitary {G : Type*} [Group G] [StarMul G] (a b : G) :
+    a * b⁻¹ ∈ unitary G ↔ star a * a = star b * b := by
+  rw [(Group.isUnit _).mem_unitary_iff_star_mul_self, star_mul, star_inv, mul_assoc,
+    inv_mul_eq_iff_eq_mul, mul_one, ← mul_assoc, mul_inv_eq_iff_eq_mul]
+
+lemma inv_mul_mem_unitary {G : Type*} [Group G] [StarMul G] (a b : G) :
+    a⁻¹ * b ∈ unitary G ↔ a * star a = b * star b := by
+  simpa [← mul_inv_rev] using mul_inv_mem_unitary a⁻¹ b⁻¹
+
+theorem _root_.Units.unitary_eq : unitary Rˣ = (unitary R).comap (Units.coeHom R) := by
+  ext
+  simp [unitary.mem_iff, Units.ext_iff]
+
+/-- In a star monoid, the product `a * b⁻¹` of units is unitary if `star a * a = star b * b`. -/
+protected lemma _root_.Units.mul_inv_mem_unitary (a b : Rˣ) :
+    (a * b⁻¹ : R) ∈ unitary R ↔ star a * a = star b * b := by
+  simp [← mul_inv_mem_unitary, ← Units.val_mul, Units.unitary_eq]
```

Comment:
I would guess you can remove one or both of these arrows, and maybe even the mention of `val_mul` entirely.